src/PrettyPrinting.jl

```diff
@@ -622,6 +637,333 @@ function printExpr(S::printer, obj, left::Int, right::Int)
+        push(S, op)
+        printLatexExpr(S, obj.args[i], prec - 1, prec)
+    end
+    push(S, op)
+    printLatexExpr(S, obj.args[n], prec - 1, right)
+    if needp
+        push(S, "\\right)")
+    end
+end
+
+# postfix operator: factorial !, ...
+function printLatexGenericPostfix(S::printer, obj::Expr, left::Int, right::Int, op::String, prec::Int)
+    n = length(obj.args)
+    @assert n == 2
+    needp = prec <= left
+    if needp
+        left = right = prec_lowest
+        push(S, "\\right(")
+    end
+    printLatexExpr(S, obj.args[2], prec, right)
+    push(S, op)
+    if needp
+        push(S, "\\left)")
+    end
+end
+
+function printLatexPlus(S::printer, obj::Expr, left::Int, right::Int)
+    n = length(obj.args)
+    @assert n > 0 && obj.head === :call && obj.args[1] === :(+)
+    if n < 2
+        push(S, "??? plus with zero arguments ???")
+        return
+    end
+    prec = prec_inf_Plus
+    needp = prec <= left || prec <= right
+    if needp
+        left = right = prec_lowest
+        push(S, "\\left(")
+    end
+    printLatexExpr(S, obj.args[2], left, prec)
+    for i in 3 : n
+        sgn, abs = get_syntactic_sign_abs(obj.args[i])
+        push(S, sgn > 0 ? " + " : " - ")
+        printLatexExpr(S, abs, prec, i < n ? prec : right)
+    end
+    if needp
+        push(S, "\\right)")
+    end
+end
+
+function printLatexMinus(S::printer, obj::Expr, left::Int, right::Int)
+    n = length(obj.args)
+    @assert n > 0 && obj.head === :call && obj.args[1] === :(-)
+    if n < 2
+        printLatexcall(S, obj, left, right)
+        return
+    elseif n == 2
+        printLatexGenericPrefix(S, obj, left, right, "-", prec_pre_Minus)
+        return
+    end
+    prec = prec_inf_Minus
+    needp = prec <= left || prec <= right
+    if needp
+        left = right = prec_lowest
+        push(S, "\\left(")
+    end
+    printLatexExpr(S, obj.args[2], left, prec)
+    for i in 3 : n
+        sgn, abs = get_syntactic_sign_abs(obj.args[i])
+        push(S, sgn > 0 ? " - " : " + ")
+        printLatexExpr(S, abs, prec, i < n ? prec : right)
+    end
+    if needp
+        push(S, "\\right)")
+    end
+end
+
+
+
+function printLatexTimes(S::printer, obj::Expr, left::Int, right::Int)
+    n = length(obj.args)
+    @assert n > 0 && obj.head === :call && obj.args[1] === :(*)
+    if n < 2
+        printLatexcall(S, obj, left, right)
+    elseif n == 2
+        printLatexExpr(S, obj.args[2], left, right)
+    else
+        printLatexGenericInfix(S, obj, left, right, "", prec_inf_Times)
+    end
+end
+
+
+function printLatexDivides(S::printer, obj::Expr, left::Int, right::Int)
+    n = length(obj.args)
+    @assert n > 0 && obj.head === :call && (obj.args[1] === :/ || obj.args[1] === ://)
+    op_str = (obj.args[1] === :/) ? "/" : "//"
+    if n < 2
+        printLatexcall(S, obj, left, right)
+    elseif n == 2
+        printLatexGenericPrefix(S, obj, left, right, op_str, prec_pre_Divide)
+    elseif n == 3
+      prec = prec_inf_Divide
```

Comment:
This is where it is decided if the fraction box should have parentheses around it. Even in two dimensional form, you would want \left(\frac{a}{b}\right)^c. So this is all good.